### PR TITLE
[Infoblox NIOS] Update timestamp parsing logic

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
There are 2 style of log formats (regards to how timestamps are handled).
1. `<29>Mar 21 09:53:51 infoblox.localdomain httpd[]: 2022-03-18 13:24:41.705Z [admin]: ........`
2. `<30>Mar 27 08:32:59 infoblox.localdomain dhcpd[2567]: ............`

**Event 1**: Contains 2 timestamps inside the log. In current default pipeline, the first timestamp `Mar 21 09:53:51` is set as `event.created` and then individual pipelines sets second timestamp as `@timestamp` if present.
**Event 2**: Contains 1 timestamp inside the log. In current default pipeline, the only timestamp `Mar 27 08:32:59` is set as `event.created` and a [set processor](https://github.com/elastic/integrations/blob/main/packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/default.yml#L78-L81) copies that into `@timestamp` if `@timestamp` is not already present from individual pipelines.

But, when Logstash is involved in between Agent and Elasticsearch, it adds `@timestamp` to the event before ingest pipeline. Hence, the logs format 2, i.e., containing single timestamp, skips the set processor leading to incorrect `@timestamp` values and also not conforming to ECS norm [@timestamp < event.created < event.ingested](https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-created:~:text=%40timestamp%20%3C%20event.created%20%3C%20event.ingested).

This PR fixes the problem by:
1. Modifying individual pipelines to not set `@timestamp` inside them.
2. Set `override: true` in the default pipeline, thus overriding the value of `@timestamp` with event's timestamp.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

`elastic-package stack down && elastic-package build && elastic-package stack up --version=8.11.0 -d -v  --services=elasticsearch && eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate -v`
```
Run pipeline tests for the package
--- Test results for package: infoblox_nios - START ---
╭───────────────┬─────────────┬───────────┬────────────────┬────────┬──────────────╮
│ PACKAGE       │ DATA STREAM │ TEST TYPE │ TEST NAME      │ RESULT │ TIME ELAPSED │
├───────────────┼─────────────┼───────────┼────────────────┼────────┼──────────────┤
│ infoblox_nios │ log         │ pipeline  │ test-audit.log │ PASS   │  12.847542ms │
│ infoblox_nios │ log         │ pipeline  │ test-dhcp.log  │ PASS   │  16.714875ms │
│ infoblox_nios │ log         │ pipeline  │ test-dns.log   │ PASS   │  10.268166ms │
╰───────────────┴─────────────┴───────────┴────────────────┴────────┴──────────────╯
--- Test results for package: infoblox_nios - END   ---
Done
```

### CODE REVIEW

File: packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -22,24 +22,21 @@ processors:
...
-      target_field: event.created
       formats:
         - MMM  d HH:mm:ss
         - MMM dd HH:mm:ss
         - MMM d HH:mm:ss
         - dd-MMM-yyyy HH:mm:ss.SSS
       on_failure:
-        - remove:
-            field: event.created
-            ignore_missing: true
```
Comment (efd6) [inline packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:37]: Should this deletion also happen in the next date processor?

**Review by efd6 — COMMENTED** [review]
AFAICS the ordering of the timestamps is coming from the structure of the event. Is this the case? If so, I think it would be helpful to make this clear in the commit message. If not, can you explain the logic more clearly?

File: packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (kcreddy) [reply packages/infoblox_nios/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: I've made logic bit clearer and this is irrelevant now.

**Comment by kcreddy** [comment]
> AFAICS the ordering of the timestamps is coming from the structure of the event. Is this the case? If so, I think it would be helpful to make this clear in the commit message. If not, can you explain the logic more clearly?

I've commit message more clearer. Let me know if that explains the issue. Thanks!

**Review by efd6 — APPROVED** [review]

**Pull request merged**